darwin/dataset/remote_dataset.py

```diff
@@ -330,6 +339,97 @@ def get_release(self, name: str = "latest"):
                 return release
         raise NotFound(self.identifier)
 
+    def split(
+        self,
+        val_percentage: float = 0.1,
+        test_percentage: float = 0,
+        force_resplit: bool = False,
+        split_seed: int = 0,
+        make_default_split: bool = True,
```

Comment:
These defaults values here make a split not being generated silently, which was unexpected to both me and @simedw when we first used the function.
I would, if you agree (later):
1. Generate a split every time (splits may have a uuid)
2. The default split is just a link to the latest split

@almazan you also mentioned we have a bug when generating splits with same seed? Could you elaborate more on how to replicate the issue?

Reply:
how about adding the export name/version to the split name, so that when we download a new version of the same dataset we get a new split.

Reply:
I don't think we need a uuid for each split, since the percentages and the seed are enough to differentiate the splits. If you want a different split using the same percentage you just need to use a different seed: it will create a new folder with the format name `split_vX_tY_sZ` where X is the training percentage, Y the test percentage, and Z the seed. If the seed is 0 then the name is `split_vX_tY`. The default split `split` always points to the latest split created unless `make_default_split` is `False`.

So what I would do is to remove `force_resplit`. I will also make sure that the split is consistent when using the same seed, since I think I saw once an unexpected behaviour.

In order to work with different dataset versions I would also add the export name to the split name, as Simon suggests.

Reply:
> how about adding the export name/version to the split name, so that when we download a new version of the same dataset we get a new split.

I just gave this solution a try and it's a bit tricky. The problem is that if we want different export versions to coexist locally, we will also need different folders for their annotations, eg `annotations_0.1` and `annotations_0.2`. Or for example, what if we use the following structure?

dataset_name:
  - images
  - 0.1:
    - annotations
    - lists
  - 0.2:
    - annotations
    - lists

Reply:
I think the images would be different to per version, you might have removed or added images. 
That's why we talked about having sym links before:

```
images/
    0.jpg
    1.jpg
    2.jpg
version/
   0.1/
        images/
              0.jpg (sym link)
              1.jpg (sym link)
       annotations/
              ....
   0.1/
        images/
              0.jpg (sym link)
              1.jpg (sym link)
              2.jpg (sym link)
       annotations/
              ....
```

Reply:
We can add those symlinks but I don't think they would be necessary since you know what images belong to a given export from the list of annotations.

Reply:
You're talking of something like: the images directory is a sort of pool, and the versions only include annotations which point to existing images? I think that should work, I can't think of any edge case, and might look cleaner that having symlinks all around
```
images/
    0.jpg
    1.jpg
    2.jpg
    3.jpg
versions/
    0.1/
        annotations/
            0.json
            1.json
    0.2/
        annotations/
            0.json
            1.json
            2.json
            3.json
```

Reply:
Yeah, exactly like that. I'll give it a try then.

Reply:
These changes are now part of a new PR: https://github.com/v7labs/darwin-py/pull/67